src/api.v2/helpers/pipeline/gem2s.js

```diff
@@ -19,17 +19,35 @@ const logger = getLogger('[Gem2sService] - ');
 
 const hookRunner = new HookRunner();
 
-const saveProcessingConfigFromGem2s = ({ experimentId, item }) => {
-  logger.log('Saving processing config for gem2s');
-  new Experiment().updateById(experimentId, { processing_config: item });
-  logger.log('Finished saving processing config for gem2s');
-};
+const continueToQC = async (payload) => {
+  const { experimentId, item } = payload;
+
+  await new Experiment().updateById(experimentId, { processing_config: item.processingConfig });
+
+  logger.log(`Experiment: ${experimentId}. Saved processing config received from gem2s`);
+
+  logger.log(`Experiment: ${experimentId}. Starting qc run because gem2s finished successfully`);
+
+  // we need to change this once we rework the pipeline message response
+  const authJWT = payload.authJWT || payload.input.authJWT;
+
+  const { stateMachineArn, executionArn } = await createQCPipeline(experimentId, [], authJWT);
+
+  await new ExperimentExecution().upsert(
+    {
+      experiment_id: experimentId,
+      pipeline_type: 'qc',
+    },
+    {
+      state_machine_arn: stateMachineArn,
+      execution_arn: executionArn,
+    },
+  );
 
-const continueToQC = () => {
+  logger.log('Started qc successfully');
 };
 
 hookRunner.register('uploadToAWS', [
```

Comment:
I know that this is not directly part of your PR, but what is the point of these hooks? Why calling `hookRunner.register` and `hookRunner.registerAll` in the middle of this file? When will these registers run? Every time we import the file? I am finding them confusing, also I notice that they are used only in this file and in `handleQCresponse`. Can you explain me what is the point of having this over-generalisation?

Reply:
they are run when the api is created only I believe

Reply:
I don't know about the reason for the hooks system compared to other options because I didn't take part in their creation.

The benefit I can see is that there were some things that had to be done when some of the step functions (qc/gem2s) steps finished, this is a good way to specify for which steps finishing something should be run (compared to using ifs this is more readable).